task/servicetest/servicetest.go

```diff
@@ -368,6 +368,23 @@ func testTaskCRUD(t *testing.T, sys *System) {
 		t.Fatalf("expected task status to be active, got %q", f.Status)
 	}
 
+	// Update task: checkpoint & last
```

Comment:
the test suite in servicetest is specifically designed to test anything that implements the `TaskService` interface. You need to write a suite of tests that tests the `Checkpointer`. I think your test suite should in the backend somewhere.